assets/models/Map.js

```diff
@@ -79,10 +88,31 @@ class Map {
     }).addTo(this.map);
   }
 
+  filterMarker(categoryIdArr = [], subCategoryIdArr = []) {
+    this.mapLayers.eachLayer(marker => {
+      this.map.removeLayer(marker);
+      if (subCategoryIdArr.length === 0 && (categoryIdArr.length === 0 || categoryIdArr.includes(marker.category_id))) {
+        this.map.addLayer(marker);
+      } else if (categoryIdArr.length === 0 && (subCategoryIdArr.length === 0 || (subCategoryIdArr.includes(marker.sub_category_id)))) {
+        this.map.addLayer(marker);
+      }
+    });
+  }
+
+  filterMarkerCount(subCategoryIdArr = []) {
+    let count = 0;
+    this.mapLayers.eachLayer(marker => {
+      if (subCategoryIdArr.includes(marker.sub_category_id)) {
+        count++;
+      }
+    });
+    return count;
+  }
+
```

Comment:
Filters are now based on markers that are present on the map rather than multiple API calls for each and every filter, Which might slow down the performance and degrade user experience